cmd/kube-apiserver/app/server.go

```diff
@@ -124,8 +124,27 @@ func Run(s *options.APIServer) error {
 		proxyDialerFn = tunneler.Dial
 	}
 
+	var proxyClientCerts []tls.Certificate
+	if s.KubeProxyClientCertFile != "" && s.KubeProxyClientKeyFile != "" {
+		clientCert, err := tls.LoadX509KeyPair(
+			s.KubeProxyClientCertFile,
+			s.KubeProxyClientKeyFile,
+		)
+
+		if err != nil {
+			glog.Fatalf("Invalid kube-proxy client certificates: %v", err)
+		} else {
+			glog.V(4).Infof("Using client certificate (%s, %s) when proxying over TLS",
+				s.KubeProxyClientCertFile, s.KubeProxyClientKeyFile)
+		}
+		proxyClientCerts = append(proxyClientCerts, clientCert)
+	}
+
 	// Proxying to pods and services is IP-based... don't expect to be able to verify the hostname
-	proxyTLSClientConfig := &tls.Config{InsecureSkipVerify: true}
+	proxyTLSClientConfig := &tls.Config{
+		InsecureSkipVerify: true,
```

Comment:
@smarterclayton the master should not expect pods to have a valid serving cert for a randomly assigned IP, but it can still be valuable for the master to prove its identity to the pod via a client cert